scripts/lint-style.py

```diff
@@ -149,6 +152,45 @@ def long_lines_check(lines, path):
             errors += [(ERR_LIN, line_nr, path)]
     return errors
 
+def indent_check(lines, path):
+    errors = []
+    indent_lvl = 0
+    in_prf = 0  # counter for nested proof blocks
+    check_rest_of_block = True  # we only check uncomplicated syntax
+    ended_with_comma = False  # track whether the previous line ends with a comma
+    inside_special = 0  # track whether we are inside ⟨⟩ or []
+    for line_nr, line in enumerate(lines, 1):
+        lstr = line.lstrip(' ')  # strip spaces from beginning of line
+        if in_prf > 0 and check_rest_of_block and ended_with_comma and not inside_special:
+            if lstr[0] == '{' and len(line) - len(lstr) != indent_lvl:
+                errors += [(WRN_IND, line_nr, path)]
+        ended_with_comma = (line.endswith(",\n"))
+        inside_special += line.count('⟨') + line.count('[') - line.count('⟩') - line.count(']')
+        if line[0] != ' ':
+            # reset the state
+            indent_lvl = 0
+            in_prf = 0
+            check_rest_of_block = True
+            ended_with_comma = False
+            inside_special = 0
+        if "match" in line or "calc" in line:
+            check_rest_of_block = False
+        if "begin" in line:
+            if line.find("begin") > 0 and in_prf == 0:
+                # complicate proof block syntax
+                check_rest_of_block = False
+            if lstr.find("begin") > 0:
+                # complicate proof block syntax
+                check_rest_of_block = False
```

Comment:
Do you have an example of a "normal" piece of Lean where the second `if` would apply that is not handled by the first `if`?

```suggestion
            # Don't check complicated proof block syntax:
            #  - a `begin` block in the middle of term mode
            if line.find("begin") > 0 and in_prf == 0:
                check_rest_of_block = False
            #  - another case where `begin` appears somewhere else in the line
            if lstr.find("begin") > 0:
                check_rest_of_block = False
```

Reply:
I think the first `if` can just be removed. I don't want to check the indentation of subproofs of the form
```
have foo : bla := begin
  stuff
end,
```
```

Reply:
Oops, nope. I clarified how they both do useful things in the other PR.